mint-client/src/clients/user.rs

```diff
@@ -150,96 +141,69 @@ impl UserClient {
         coins: Coins<SpendableCoin>,
         mut rng: R,
     ) -> Result<OutPoint, ClientError> {
-        const OUT_IDX: u64 = 0;
+        let mut tx = TransactionBuilder::default();
+
+        let (mut coin_keys, coin_input) = self.mint_client().create_coin_input_from_coins(coins)?;
+        tx.input(&mut coin_keys, Input::Mint(coin_input));
+        let txid = self
+            .submit_tx_with_change(tx, DbBatch::new(), &mut rng)
+            .await?;
+
+        Ok(OutPoint { txid, out_idx: 0 })
+    }
 
+    pub async fn pay_coins<R: RngCore + CryptoRng>(
```

Comment:
I think this is a slightly confusing name. `spend_coins` might be better?

Reply:
In the function you are spending your own coins to include the `BlindToken` as an output.

Maybe `pay_for_coins`...it's hard to think of a good name.